pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -100,12 +103,47 @@ type Config struct {
 	}
 }
 
+// ApiWithNamespace stores api and namespace in context
+type apiWithNamespace struct {
+	namespace string
+	apiCall   string
+}
+
 func init() {
+	registerMetrics()
 	cloudprovider.RegisterCloudProvider(
 		ProviderName,
 		func(config io.Reader) (cloudprovider.Interface, error) {
 			return newGCECloud(config)
 		})
+	gensupport.RegisterHook(trackAPILatency)
+}
+
+func trackAPILatency(ctx context.Context, req *http.Request) func(resp *http.Response) {
+	requestTime := time.Now()
+	t := ctx.Value("kube-api-namespace")
+	apiNamespace, ok := t.(apiWithNamespace)
+
+	if !ok {
+		return nil
+	}
+
+	apiResponseReceived := func(resp *http.Response) {
+		timeTaken := time.Since(requestTime).Seconds()
```

Comment:
According to https://prometheus.io/docs/practices/naming/#metric-names metrics should use base units. I guess we should be okay, since we are reporting seconds as floats anyways.